api: make VideoModeHandle into VideoMode (#4060)
rust-windowing/winit: src/changelog/unreleased.md

- On macOS, no longer emit `Focused` upon window creation.
- On iOS, emit more events immediately, instead of queuing them.
- Update `smol_str` to version `0.3`
- `VideoModeHandle` to `VideoMode` which stores plain data.

> - `Fullscreen::Exclusive` to take `(MonitorHandle, VideoMode)`.

Reviewer: Changed/renamed prefix for both?

> - Rename `VideoModeHandle` to `VideoMode` which stores plain data.
> - Make `Fullscreen::Exclusive` to take `(MonitorHandle, VideoMode)`.

Reviewer 2: ```suggestion
- Rename `VideoModeHandle` to `VideoMode`, now it only stores plain data.
- Make `Fullscreen::Exclusive` contain `(MonitorHandle, VideoMode)`.
```
